src/main/scala/com/codacy/client/bitbucket/v2/UserPermission.scala

```diff
@@ -0,0 +1,14 @@
+package com.codacy.client.bitbucket.v2
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+final case class UserPermission(user: SimpleUser, repository: SimpleRepository, permission: String)
+
+object UserPermission {
+  implicit val reader: Reads[UserPermission] = {
+    ((__ \ "user").read[SimpleUser] and
+      (__ \ "repository").read[SimpleRepository] and
+      (__ \ "permission").read[String])(UserPermission.apply _)
+  }
```

Comment:
Are you manual `Reads` for safety and better retro-compatibility in case of changes in play?

Reply:
@lolgab I think you'll find its more for library consistency as the existing would be done the same. @bmbferreira feel free to correct me!

Reply:
yep, @peavers is right. We are already doing this for other stuff, I'm just keeping consistency.